### ISSUE TITLE
lazygit freeze after press twice L to scroll right in commits window

### ISSUE THREAD

**Issue by jiaming-shi** [jesseduffield/lazygit#2289]
**Describe the bug**
lazygit ui freeze after I press twice `L` to scroll right in commits window.

**To Reproduce**

1. Clone a git repo to reproduce: `git clone https://github.com/jesseduffield/lazygit.git`
2. Open lazygit in cloned repo: `cd lazygit && lazygit`
3. Switch to Commits window
4. Press twice `L`

**Expected behavior**
lazygit works as usual.

**Screenshots**

**Version info:**
_Run `lazygit --version` and paste the result here_
```
commit=, build date=, build source=homebrew, version=0.36.0, os=darwin, arch=amd64, git version=2.37.1 (Apple Git-137.1)
```
_Run `git --version` and paste the result here_
```
git version 2.37.1 (Apple Git-137.1)
```

**Additional context**

**Note:** please try updating to the latest version or [manually building](https://github.com/jesseduffield/lazygit/#manual) the latest `master` to see if the issue still occurs.

**Comment by mark2185** [comment]
Could you maybe record a screencast? Which language are you using?

I'm asking because I can't reproduce it.

**Comment by jiaming-shi** [comment]
https://user-images.githubusercontent.com/11895323/203792322-99d08d84-3b5f-4d66-8dfe-7065636b98e4.mov

**Comment by mark2185** [comment]
Does it happen with a clean `.gitconfig` and `lazygit/config.yml` ?

**Comment by jiaming-shi** [comment]
After remove `~/.gitconfig` and lazygit config, still have this problem.

**Comment by jiaming-shi** [comment]
This is the output of `lazygit --logs`:
At 21:53:53, I pressed `L` first time, 21:54:02 is the second time.
After that, lazygit is frozen, but log keeps flush. Also, the cpu utilization of the lazygit process is 100%.

```
Nov 24 21:53:53 |DEBU| RunCommand
Nov 24 21:53:53 |DEBU| using cache for key status.showUntrackedFiles
Nov 24 21:53:53 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z"
Nov 24 21:53:58 |DEBU| RunCommand command="git show --submodule --color=always --unified=3 --no-renames --stat -p 924a152ae92372a178bd0fc32b411612b995a6a1"
Nov 24 21:54:02 |DEBU| RunCommand
Nov 24 21:54:03 |DEBU| using cache for key status.showUntrackedFiles
Nov 24 21:54:03 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z"
Nov 24 21:54:05 |DEBU| RunCommand command="git show --submodule --color=always --unified=3 --no-renames --stat -p 924a152ae92372a178bd0fc32b411612b995a6a1"
Nov 24 21:54:13 |DEBU| using cache for key status.showUntrackedFiles
Nov 24 21:54:13 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z"
Nov 24 21:54:23 |DEBU| using cache for key status.showUntrackedFiles
Nov 24 21:54:23 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z"
Nov 24 21:54:24 |DEBU| RunCommand command="git fetch"
Nov 24 21:54:26 |INFO| refreshing the following scopes in async mode: branches,commits,remotes,tags
Nov 24 21:54:26 |DEBU| RunCommand command="git tag --list --sort=-creatordate"
Nov 24 21:54:26 |DEBU| RunCommand command="git branch -r"
Nov 24 21:54:26 |DEBU| RunCommand command="git merge-base \"HEAD\" \"HEAD\"@{u}"
Nov 24 21:54:26 |DEBU| RunCommand command="git for-each-ref --sort=-committerdate --format=\"%(HEAD)%00%(refname:short)%00%(upstream:short)%00%(upstream:track)\" refs/heads"
Nov 24 21:54:26 |DEBU| RunCommand command="git symbolic-ref --short HEAD"
Nov 24 21:54:26 |DEBU| RunCommand command="git merge-base \"HEAD\" \"master\""
Nov 24 21:54:33 |DEBU| using cache for key status.showUntrackedFiles
Nov 24 21:54:33 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z"
Nov 24 21:54:43 |DEBU| using cache for key status.showUntrackedFiles
Nov 24 21:54:43 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z"
Nov 24 21:54:53 |DEBU| using cache for key status.showUntrackedFiles
Nov 24 21:54:53 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z"
Nov 24 21:55:03 |DEBU| using cache for key status.showUntrackedFiles
Nov 24 21:55:03 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z"
Nov 24 21:55:13 |DEBU| using cache for key status.showUntrackedFiles
Nov 24 21:55:13 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z"
Nov 24 21:55:23 |DEBU| using cache for key status.showUntrackedFiles
Nov 24 21:55:23 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z"
Nov 24 21:55:24 |DEBU| RunCommand command="git fetch"
Nov 24 21:55:26 |INFO| refreshing the following scopes in async mode: branches,commits,remotes,tags
Nov 24 21:55:26 |DEBU| RunCommand command="git branch -r"
Nov 24 21:55:26 |DEBU| RunCommand command="git tag --list --sort=-creatordate"
Nov 24 21:55:26 |DEBU| RunCommand command="git for-each-ref --sort=-committerdate --format=\"%(HEAD)%00%(refname:short)%00%(upstream:short)%00%(upstream:track)\" refs/heads"
Nov 24 21:55:33 |DEBU| using cache for key status.showUntrackedFiles
Nov 24 21:55:33 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z"
```

**Comment by mark2185** [comment]
@leiswatch you reacted with :+1: to the original issue, do you have the same issue?

**Comment by Ryooooooga** [comment]
Please try this branch: https://github.com/jesseduffield/lazygit/pull/2276

**Comment by leiswatch** [comment]
@mark2185 Yes, I have the same issue. I open lazygit, go to `Local branches` and I hit `L` twice. Lazygit freezes.

`commit=d1a8b05401ccda1e52215203139a252eb3be3b79, build date=2022-11-14T09:22:31Z, build source=binaryRelease, version=0.36.0, os=linux, arch=amd64, git version=2.38.1`

`lazygit --logs`

`Nov 25 15:41:02 |DEBU| RunCommand command="git branch -r"
Nov 25 15:41:02 |DEBU| RunCommand command="git merge-base \"HEAD\" \"HEAD\"@{u}"
Nov 25 15:41:02 |DEBU| RunCommand command="git for-each-ref --sort=-committerdate --format=\"%(HEAD)%00%(refname:short)%00%(upstream:short)%00%(upstream:track)\" refs/heads"
Nov 25 15:41:02 |DEBU| RunCommand command="git log --graph --color=always --abbrev-commit --decorate --date=relative --pretty=medium refs/heads/develop --"
Nov 25 15:41:02 |DEBU| RunCommand command="git symbolic-ref --short HEAD"
Nov 25 15:41:02 |DEBU| RunCommand command="git merge-base \"HEAD\" \"master\""
Nov 25 15:41:02 |ERRO| fatal: Not a valid object name master
 
Nov 25 15:41:03 |DEBU| RunCommand command="git log --graph --color=always --abbrev-commit --decorate --date=relative --pretty=medium refs/heads/develop --"
Nov 25 15:41:10 |DEBU| using cache for key status.showUntrackedFiles 
Nov 25 15:41:10 |DEBU| RunCommand command="git status --untracked-files=all --porcelain -z"`

I'm running PopOS `Linux pop-os 6.0.6-76060006-generic #202210290932~1667401208~22.04~d2df702 SMP PREEMPT_DYNAMIC Wed N x86_64 x86_64 x86_64 GNU/Linux`

**Comment by jiaming-shi** [comment]
> Please try this branch: #2276

Build from this branch, lazygit will not freeze anymore.

**Comment by uloco** [comment]
I guess it breaks when you cannot scroll more vertically. I have it for example in the stashes window, when I press L until it cannot scroll anymore it freezes.

**Comment by mark2185** [comment]
@uloco does it happen on latest `master`? #2276, which was supposed to fix this issue, was merged almost a month ago.

**Comment by uloco** [comment]
I have only tested what is released on brew latest. Probably fixed already...

**Comment by x3rAx** [comment]
I also have this issue. For me it's not exactly that you have to press L twice but rather it happens every time when there is nothing more to scroll to the right. So it also happens in the "Files" view or in "Unstaged Changes". In the repo I tried it, there is a very long commit message so I was able to press "L" 4 times without freeze, but when I pressed it a 5th time, lazygit freezes since it can't scroll any further.

I can confirm that it is fixed on `master` but now it seems like I can scroll infinitely far to the right even in views where there is nothing to scroll right to.

**Comment by madjxatw** [comment]
Same issue confirmed here. Wondering whether this issue would be fixed in next stable release?

```
commit=, build date=, build source=unknown, version=0.36.0, os=linux, arch=amd64, git version=2.39.1
```

**Comment by mark2185** [comment]
> Wondering whether this issue would be fixed in next stable release?

Yes. 

Until then, build from source.

**Comment by mark2185** [comment]
Fixed in [v0.37.0](https://github.com/jesseduffield/lazygit/releases/tag/v0.37.0).
